lua/nvim-tree/renderer.lua

```diff
@@ -296,6 +299,19 @@ local function update_draw_data(tree, depth, markers)
         table.insert(hl, {'NvimTreeImageFile', index, offset+#icon+#git_icons, -1 })
       end
 
+      if vim.g.nvim_tree_highlight_icons then
+        if vim.fn.bufloaded(node.absolute_path) > 0 then
+          table.insert(hl, {'NvimTreeOpenedFile', index, offset, offset+#icon })
+        else
+          for i, value in pairs(hl) do
+            if value[1] == 'NvimTreeOpenedFile' and value[2] == index then
+              table.remove(hl, i)
+              break
```

Comment:
is this loop required ? i don't think this highlight would stay as it's regenerated each time :)

Reply:
You're right! Thanks! :+1: :rocket: